frontend/src/app/login-form/login-form.component.spec.ts

```diff
@@ -1,25 +1,58 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {
+  async,
+  TestBed,
+  getTestBed,
+  fakeAsync,
+  tick,
+} from '@angular/core/testing';
 
 import { LoginFormComponent } from './login-form.component';
+import { AppMaterialModule } from '../app-material/app-material.module';
+import { HttpService, LoginResponse } from '../services/http.service';
+import { of, asyncScheduler, throwError } from 'rxjs';
 
 describe('LoginFormComponent', () => {
+  let injector: TestBed;
   let component: LoginFormComponent;
-  let fixture: ComponentFixture<LoginFormComponent>;
+  let httpServiceMock: jasmine.SpyObj<HttpService>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ LoginFormComponent ]
-    })
-    .compileComponents();
+      declarations: [LoginFormComponent],
+      imports: [AppMaterialModule],
+      providers: [
+        LoginFormComponent,
```

Comment:
Component is not a provider :D

Reply:
According to official [documentation](https://angular.io/guide/testing-components-basics), I think it is necessary.
Code sample from docs:
```
beforeEach(() => {
  TestBed.configureTestingModule({
    // provide the component-under-test and dependent service
    providers: [
      WelcomeComponent,
      { provide: UserService, useClass: MockUserService }
    ]
  });
  // inject both the component and the dependent service.
  comp = TestBed.inject(WelcomeComponent);
  userService = TestBed.inject(UserService);
});
```
Removing `LoginFormComponent` from the provider list broked some UTs

Reply:
Noice. Didn't know that